src/Pact/Native/Db.hs

```diff
@@ -220,6 +237,38 @@ read' g0 i as@(table@TTable {}:TLitString key:rest) = do
 
 read' _ i as = argsError i as
 
+
+foldDB' :: NativeFun e
+foldDB' i [tbl, TApp qry _, TApp consumer _] = do
+  table <- reduce tbl >>= \case
+    t@TTable{} -> return t
+    t -> evalError' i $ "Expected table as first argument to foldDB, got: " <> pretty t
+  g0 <- computeGas (Right i) (GUnreduced [])
+  (!g1, ks) <- getKeys g0 table
+  (!g2, xs) <- foldlM (fdb table) (g1, []) ks
+  pure (g2, TList (V.fromList (reverse xs)) TyAny def)
+  where
+  asBool (TLiteral (LBool satisfies) _) = return satisfies
+  asBool t = evalError' i $ "Unexpected return value from fold-db query condition " <> pretty t
+  getKeys g table = gasPostReads i g sort $ do
```

Comment:
I don't think we need `gasPostReads` here? Also we should consider forking to sorted `keys` across the board

Reply:
:+1: